### ISSUE TITLE
 Show all articles of a journal on a timeline

### ISSUE THREAD

**Issue by nuest** [TIBHannover/optimetaGeo#74]
Split out of #7

**Comment by nuest** [comment]
https://arkeopen.org/?lng=7.74727&lat=50.48248&z=6.91&p=left&rp=site%C2%A71311142&c=2037&sd=-2199&ed=-1350 uses a chronological system for the UI/interaction, but the actual datasets seem to have actual timestamps, though range seems to be capped at -5000.

**Comment by nuest** [comment]
https://community.esri.com/t5/arcgis-javascript-maps-sdk-blog/timeslider-widgets-for-historical-or-geological/ba-p/1077255

**Comment by nuest** [comment]
Pangaea has regular time stamps for filtering, but going back only to 1923, so it seems to be the publication date?

https://www.pangaea.de/?t=Paleontology&mindate=1923-11-01&maxdate=1964-11-18

BUT I can add other dates manually: https://www.pangaea.de/?t=Paleontology&mindate=0000-11-01&maxdate=1900-11-18

So there are datasets with older data: https://doi.pangaea.de/10.1594/PANGAEA.865056

![image](https://github.com/TIBHannover/ojsGeo/assets/1325054/500254ba-b1ef-4591-bdac-f24d1fa866a4)

**Comment by nuest** [comment]
Leaflet Plugins are also not very exhaustive:

- https://svitkin.github.io/leaflet-timeline-slider/
- https://github.com/svitkin/leaflet-timeline-slider
- https://github.com/skeate/Leaflet.timeline
- https://skeate.dev/Leaflet.timeline/
- https://github.com/zimmicz/Leaflet-Timeline-Control

**Comment by tomniers** [comment]
potential library for implementation: https://visjs.github.io/vis-timeline/examples/timeline/

**Comment by nuest** [comment]
Revisiting the candidates and casting a wider net (April 2026). Summary:

### Libraries re-checked

| Library | License | Maint. (12mo) | Ranges | Deep time (BCE) | Notes |
|---|---|---|---|---|---|
| [vis-timeline](https://github.com/visjs/vis-timeline) v8.5.0 | Apache-2.0 / MIT | 161+ commits | yes (`range` items) | workable — negative years via expanded ISO `-000050-…`; the demo at https://visjs.github.io/vis-timeline/examples/timeline/ scrolls to ~-100000 smoothly | built-in item clustering; peer-depends on Moment (already shipped by the plugin) |
| [TimelineJS3](https://github.com/NUKnightLab/TimelineJS3) v3.9.11 | MPL-2.0 | 53 commits | yes | **best-in-class** — first-class `era`/`epoch`/`eon` scales | storytelling UX (slide-per-event); scales poorly past ~100 items |
| [Timelines Chart (vasturiano)](https://github.com/vasturiano/timelines-chart) v2.14.3 | MIT | light but responsive | yes (swimlane segments) | year-0..9999 only (d3-scale + Date) | cleanest swimlane; ~30 KB gzip |
| [leaflet-timeline-slider](https://github.com/svitkin/leaflet-timeline-slider) | MIT | stale 5+ yrs | no | – | jQuery-coupled; avoid |
| [Leaflet-Timeline-Control](https://github.com/zimmicz/Leaflet-Timeline-Control) | ISC | stale 5+ yrs | no | – | avoid |
| [Leaflet.timeline (skeate)](https://github.com/skeate/Leaflet.timeline) v1.4.3 | ISC | dormant 12mo | yes | yes (numeric) | **different purpose** — animates GeoJSON over time on a map; complement, not replacement |
| [Apache ECharts](https://github.com/apache/echarts) 6.x | Apache-2.0 | daily | via `custom` series | `value`-axis workaround for BCE | heavy (~95 KB gzip); overkill for one view |
| [d3-milestones](https://github.com/walterra/d3-milestones) v1.6.0 | Apache-2.0 | active | points only | d3-time-format year-0 issue | elegant auto-aggregation if one date per article is acceptable |
| [Frappe Gantt](https://github.com/frappe/gantt) v1.2.2 | MIT | active | yes | year-0 floor | PM feel; wrong UX shape |
| [react-calendar-timeline](https://github.com/namespace-ee/react-calendar-timeline) | MIT | active | yes | – | React-only → out of scope |

### Bespoke alternatives evaluated

- **Plain HTML/CSS + ~80 LOC JS** (absolutely-positioned `<div>`s, CSS `calc()` on custom properties): under a day; numeric-year math so deep time is free; a11y/i18n essentially free because each period is a real focusable DOM node.
- **D3 hand-roll** (`d3-scale` + `d3-axis` + `d3-selection`, ~30 KB gzip tree-shaken): `scaleLinear().domain([-8000, 2024])` supports deep time natively; full control over per-rect ARIA semantics and event bridges to Leaflet.
- **Observable Plot / Vega-Lite**: 150–270 KB gzip; their temporal scales are `Date`-backed (same BCE failure), so you'd degrade to quantitative anyway — no win over D3.
- **Leaflet-with-CRS.Simple**: a second map, not a reuse of the existing one (our maps use EPSG:3857); reuses API familiarity but bad axis semantics and weak a11y.

### Recommendation

**vis-timeline** — best practical fit. Aligned with the plugin's existing Moment dep, true range items, built-in clustering for hundreds of items, active maintenance, locale-aware, and the deep-time escape hatch (expanded-year ISO) is documented and working on their own demo. Trade-off accepted: axis tick aesthetics degrade past ~10 kyr (vis-timeline issue [#162](https://github.com/visjs/vis-timeline/issues/162)), which is acceptable because the main target is modern observational data; archaeology/paleontology journals remain readable with coarser ticks.

TimelineJS3 stays in reserve if a journal specifically needs geological/archaeological era labelling; Timelines Chart stays in reserve as the cleanest swimlane look if the final UI moves in that direction.

### Deep-time gotcha (applies to any choice)

JS `Date` is unreliable for years 0–99 and pre-1582-Gregorian. Any library backed by `d3-time`, Vega temporal type, or a `time` axis inherits this. Numeric/quantitative scales are the only safe path for full deep time; vis-timeline's expanded-year ISO string is the library-specific workaround.

Implementation plan follows in a Phase-2 ticket. Research was Claude-assisted; the full UX survey of comparable scholarly portals (Pangaea, Arkeopen, Neotoma, Europeana, …) was blocked by sandbox permissions and is still open — happy to run it if someone wants the UI-pattern comparison before committing.

**Comment by nuest** [comment]
Fallback to note, in case vis-timeline's deep-time behaviour turns out to be flakier in practice than its demo suggests: the data shape \`JournalMapHandler\` will emit for the timeline strip is library-agnostic (just \`{publicationId, title, authors, issue, temporal}\` per article). A plain HTML/CSS swimlane — each period an absolutely-positioned \`<div>\` on a \`position: relative\` strip, widths via \`calc()\` on CSS custom properties — is a ~1 day build and handles BCE natively (everything is a JS number). Swapping renderers later would be a single-file change in \`js/journal.js\` / \`js/issue.js\`.

Not planning to build this; noting it here so future-us knows the escape hatch exists if vis-timeline surprises us.
